File: testdata/bench/javascript/BenchmarkTest00010.js
// Safe: SQL injection prevented via knex.raw with bindings
const express = require('express');
const knex = require('./knex');
const router = express.Router();
router.post('/test010', (req, res) => {
    const param = req.body.filter;
    knex.raw("SELECT * FROM orders WHERE status = ?", [param])
        .then(results => res.json(results))
        .catch(err => res.status(500).json({ error: 'Query failed' }));
});
module.exports = router;

File: testdata/bench/rust/BenchmarkTest00010.rs
use actix_web::{web, HttpResponse};

// Safe: SQL with hardcoded value, no user input in query
async fn handler(path: web::Path<String>) -> HttpResponse {
    let _param = path.into_inner();
    let id = "admin";
    let sql = format!("SELECT * FROM users WHERE role = '{}'", id);
    // sqlx::query(&sql).execute(&pool).await
    HttpResponse::Ok().body("safe")
}
